python/mxnet/gluon/block.py

```diff
@@ -48,8 +48,9 @@ class _BlockScope(object):
     def __init__(self, block):
         self._block = block
         self._counter = {}
-        self._old_scope = None
-        self._name_scope = None
+        self._local = threading.local()
```

Comment:
How about using contextvars here? This will ensure the code is compatible with python coroutines.
https://docs.python.org/3/library/contextvars.html

You need to include https://pypi.org/project/contextvars/ as dependency on Python 3.5, 3.6

Reply:
Thanks for sharing this. Do we have a plan to switch to contextvars in the near future? This pr focuses on the _BlockScope issue. Maybe we can switch all threading.local() to contextvars in another pr.

Reply:
It's a bug. MXNet will currently experience undefined behaviour when coroutines are used. There's no immediate plan to fix it, as it requires some backend work as well. However, new code should ideally be written in a correct way to reduce future efforts.

Reply:
Considering that additional dependency on contextvars needs to be introduced for py3.5, a separate pr for contextvars seems better, which adds the dependency, and switch all thread local to contextvars. I may still go with thread local here, but I’ll be glad to switch to contextvars after we add the dependency.